### DISCUSSION TITLE
[6일차-지원] 리액트의 생명주기에 대해 말해보세요.

### DISCUSSION THREAD

**Discussion by smarfy99** [zzangCS/zzangInterview#23] [프론트 면접 질문]
클래스 컴포넌트와 연결지어서

**Comment by ez1n** [comment]
- **Life Cycle**

> 컴포넌트가 생성되고 사용되고 소멸되기까지의 과정 (컴포넌트 생명주기)

 ⇒ mount, update, unmount로 구분
    
- **호출되는 메소드 (method)**
1. componentDidMount() : 최초로 컴포넌트가 생성된 후(첫 번째 렌더링 마친 후) 한 번 호출
2. componentDidUpdate() : 컴포넌트의 속성 값 또는 상태값이 변경될 때 호출 (컴포넌트 업데이트 된 후)
3. componentWillUnmount() : 컴포넌트가 소멸될 때 호출
4. render() : 초기에 화면을 그려줄 때와 업데이트될 때 호출

**Reply by ez1n** [reply]
https://www.zerocho.com/category/React/post/579b5ec26958781500ed9955

**Comment by smarfy99** [comment]
<img width="782" alt="image" src="https://user-images.githubusercontent.com/82459123/227176504-3b177f71-4e7b-46d5-9848-d9cd9e126e05.png">
1. Mount
- componentWillMount메소드 호출

- render로 컴포넌트를 DOM에 부착

- Mount가 완료되면 componentDidMount - DOM 접근
2. Props Update
- componentWillReceiveProps 메소드
- shouldComponentUpdate, componentWillUpdate
- render
- componentDidUpdate
3. State Update
- shouldComponentUpdate
- componentWillUpdate
- render
- componentDidUpdate
4. Unmount
- componentWillUnmount

**Reply by smarfy99** [reply]
https://www.zerocho.com/category/React/post/579b5ec26958781500ed9955

**Comment by skyhanull** [comment]
Mount(탄생) > Update, Re-render(변화) > Unmount(죽음)

componentDidMount : 컴포넌트를 만들고, 첫렌더링을 다 마친 후 실행합니다.(useEffect빈배열시)

componentDidUpdate: 컴포넌트가 리렌더링 될 때, 즉 render()가 업데이트 될 때마다 실행됩니다.

componentWillUnMount : 컴포넌트를 dom에서 제거할 때 실행합니다.(cleanup함수)
